lib/templates.ts

```diff
@@ -604,27 +604,38 @@ export async function updateIsPublishedForTemplate(
   publishFormType: string,
   publishDescription: string
 ): Promise<FormRecord | null> {
+  // Alias the isPublished value to newPublishStatus for clarity within the function
+  const newPublishStatus = isPublished;
+
   const { user } = await authorization.canPublishForm(formID).catch((e) => {
     logEvent(e.user.id, { type: "Form", id: formID }, "AccessDenied", "Attempted to publish form");
     throw e;
   });
 
   // Delete all form responses created during draft mode
-  if (isPublished && process.env.APP_ENV !== "test") {
-    await deleteDraftFormResponses(formID);
+  if (newPublishStatus && process.env.APP_ENV !== "test") {
+    try {
+      await deleteDraftFormResponses(formID);
+    } catch (e) {
+      if (e instanceof TemplateAlreadyPublishedError) {
+        // Already published, so we can just return the full template
+        return getFullTemplateByID(formID);
+      }
+
+      throw e;
+    }
   }
 
-  // We use a where unique input to ensure we are only updating the form if it is not published
```

Comment:
Removed this 

> ensure we are only updating the form if it is not published

False can be passed in as the param which would also be an update